packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -1395,6 +1396,17 @@ describe('CozyClient', () => {
       expect(client.requestQuery).not.toHaveBeenCalled()
     })
 
+    it('should not fetch policy when background fetching enabled', async () => {
+      jest.spyOn(client, 'requestQuery')
+      const fetchPolicy = jest.fn(() => false)
+      await client.query(query, {
+        as: 'allTodos',
+        fetchPolicy,
+        backgroundFetching: true
+      })
+      expect(fetchPolicy).not.toHaveBeenCalled()
+    })
```

Comment:
I'm really not sure about that. `fetch policy` is useful for caching and to avoid fetch "spam" when rendering several components. 

`background fetching` will be more used for refreshing data when they already exists. 

It should not be one or the other. We should find a way to make them live together. 

Maybe that, by default, background fetching should respect the fetch policy. If it returns false, then we should not run the query. 

After that, we can create a plugin to force the background fetch, for instance, when giving the focus or something. 

What is your opinion about this feature?

Reply:
After reconsideration:

I see 3 different features that can be totally independant:

- **fetchPolicies**, exactly what you define above

- **backgroundFetching** that handles queryState.fetchStatus + isFetching, to prevent the Spinner and the bad UX. Implemented in this PR, if I revert that change  line 917 in CozyClient.js and update this test    `expect(fetchPolicy).toHaveBeenCalled()`

- **forcingRefetch** - or something called [refetchOnWindowFocus in reactQuery](https://tanstack.com/query/v4/docs/guides/window-focus-refetching?from=reactQueryV3&original=https%3A%2F%2Freact-query-v3.tanstack.com%2Fguides%2Fwindow-focus-refetching), when giving the focus, when an app needs to refetch whatever the fetch policies.


I am pretty sure, the background fetching will make the fetchPolicies decrease. Maybe the `forcingRefetch` is not that urgent. Personally I think we should try to see how the 2 first features will collaborate, after considering implementing the `forcingRefetch`.

So I agree 100% with you.